src/map/instance.cpp

```diff
@@ -66,240 +82,251 @@ struct instance_db *instance_searchname_db(const char *instance_name) {
  * @param sd: Player data to attach
  * @param target: Target display type
  */
-void instance_getsd(unsigned short instance_id, struct map_session_data **sd, enum send_target *target) {
-	switch(instance_data[instance_id].mode) {
+void instance_getsd(uint16 instance_id, struct map_session_data **sd, enum send_target *target) {
+	auto idata = instance_search(instance_id);
```

Comment:
this could return nullptr no ?
@return shared_ptr of instances on success or nullptr
Well technically it always return a shared_ptr but with no raw_pointer associated.